honeycombio/resource_dataset.go

```diff
@@ -23,19 +23,39 @@ func newDataset() *schema.Resource {
 				ForceNew:     true,
 				ValidateFunc: validation.StringLenBetween(1, 255),
 			},
+			"description": {
+				Type:         schema.TypeString,
+				Computed:     false,
+				Required:     false,
+				Optional:     true,
+				ForceNew:     false,
+				ValidateFunc: validation.StringLenBetween(0, 1024),
+			},
 			"slug": {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+			"expand_json_depth": {
+				Type:         schema.TypeInt,
+				Computed:     false,
+				Required:     false,
+				Optional:     true,
+				ForceNew:     false,
+				ValidateFunc: validation.IntBetween(0, 10),
+			},
 		},
 	}
 }
 
 func resourceDatasetCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*honeycombio.Client)
 
+	description := d.Get("description").(string)
+	expandJSONDepth := d.Get("expand_json_depth").(int)
```

Comment:
In the name of consistency, I think we should move this into the struct initialization or bring `name` out of the struct initialization